### ISSUE TITLE
Switch buffer arrows wont work if a pattern is entered

### ISSUE THREAD

**Issue by jbharat** [abo-abo/swiper#1159]
While switching to buffer in ivy, if no pattern is entered, arrows (ctrl j/k) work perfectly. When a pattern is entered, arrows (ctrl j/k) stops working.  

This behaviour does not occur with searching(swiper), dired. Only happens with switching buffer.

This started today. I searched for an existing issue for this but i didnt find any or didnt understand if they were the same. Let me know if there is a bug already for this. 

package -
ivy-20170813.351

**Comment by abo-abo** [comment]
Can you reproduce with `emacs -Q`?

**Comment by dieggsy** [comment]
Huh, I'm actually experiencing this too, though I haven't been able to reproduce with `emacs -Q` yet

**Comment by Stebalien** [comment]
Same here on spacemacs. I'll try to reproduce with a minimal setup when I have time.

**Comment by jbharat** [comment]
I am using spacemacs as well. I was not able to reproduce with `emacs -Q`

**Comment by abo-abo** [comment]
@jbharat Then next step after `emacs -Q` is `emacs -Q -l init.el`; you put only your ivy-specific config in `init.el`.

**Comment by jbharat** [comment]
@abo-abo I tested it with stock emacs. Buffer switch works fine(with/without pattern). Seems to be issue with spacemacs. Should I close this issue?

**Comment by dieggsy** [comment]
@jbharat I am not using spacemacs, so it may be an issue with evil...

**Comment by abo-abo** [comment]
> Should I close this?

Let's wait until someone posts a reproducible scenario. The issue affects many people.

**Comment by Dickby** [comment]
I have this problem too, it seems to be triggered by setting ivy-regex-builder-alist to ivy-regex-ignore-order (probably related to #1163)

**Comment by jbharat** [comment]
@Dickby I have tested by removing all additional config of ivy. I have just ivy layer added to my .spacemacs file. I am still facing this issue. Not sure if it is ivy or some other package.

**Comment by abo-abo** [comment]
@Dickby thanks for the pointer, please test.

**Comment by dieggsy** [comment]
@abo-abo i _think_ that fixed some of it, though I think if I enter text for which there is an exact match (but also other matches) I still can't move the arrow keys. 

e.g. (`|` marks cursor
```
Switch-to-buffer: thing|
  thing1
  thing2
  thing
```
The selection will remain at `thing` without being able to move unless I either type less than the exact match, or add a space.

**Comment by abo-abo** [comment]
@dieggsy thanks for the pointer, please test.

**Comment by jbharat** [comment]
working for me. Thanks @abo-abo @dieggsy

**Comment by zzjjzzgggg** [comment]
When I search a string containing white space, e.g., "aaa bbb", ivy just matches "aaa", and doesn't match "bbb".

**Comment by jbharat** [comment]
@zzjjzzgggg if you are using fuzzy, just use aaabbb. I think it will match it.

**Comment by zzjjzzgggg** [comment]
I am not using fuzzy. Searches containing white spaces no longer work after the recent two commits.

**Comment by dieggsy** [comment]
@abo-abo your commit seemed to point out the latest issue I pointed out, however I can now confirm @zzjjzzgggg 's described behavior as well (that is, anything after a space now does nothing to further narrow candidates.)

**Comment by jbharat** [comment]
Yep pattern after space is ignored. Tested

**Comment by abo-abo** [comment]
Please try again.

For minibuffer highlights when using `swiper`, add e.g.

```elisp
(add-to-list 'ivy-highlight-functions-alist
             '(swiper--re-builder . ivy--highlight-ignore-order))
```

**Comment by Dickby** [comment]
looks good! fixes #1163 aswell.
